src/ArrayComparator.php

```diff
@@ -53,8 +53,16 @@ public function accepts($expected, $actual)
     public function assertEquals($expected, $actual, $delta = 0.0, $canonicalize = false, $ignoreCase = false, array &$processed = [])/*: void*/
     {
         if ($canonicalize) {
-            sort($expected);
-            sort($actual);
+            // If the expected array has non-integer keys, treat arrays as being associative
```

Comment:
if `count($expected, COUNT_RECURSIVE) !== count($actual, COUNT_RECURSIVE)`, then we can skip sorting